lib/elixir/test/elixir/string_io_test.exs

```diff
@@ -140,9 +140,13 @@ defmodule StringIOTest do
   test "IO.write with non-printable arguments" do
     {:ok, pid} = StringIO.open("")
 
-    assert_raise ArgumentError, fn ->
+    assert_raise ArgumentError, ~r/not valid character data/, fn ->
       IO.write(pid, [<<1::1>>])
     end
+
+    assert_raise ArgumentError, ~r/could not convert argument encoding/, fn ->
```

Comment:
FYI:

When using a normal erlang file instead of `StringIO`, the current error looks like this:

```
** (ArgumentError) errors were found at the given arguments: unknown error: {put_chars,unicode,<<"Þ">>}
    (stdlib 3.16.1) io.erl:99: :io.put_chars(:standard_io, <<222>>)
```

After the release of the Erlang PR, it will look like this:

```
** (ArgumentError) errors were found at the given arguments:

  * 2nd argument: not valid character data (an iodata term)

    (stdlib 3.17.1) io.erl:99: :io.put_chars(:standard_io, <<222>>)
    iex:1: (file)
```